File: src/install_logger.nim
## install_logger

import strutils
import times

const logfile = "Log3.md"

var log = try: readFile(logfile) except: "|Timestamp|Software|\n|-|-|\n"

echo "Enter program name to log it. Leave blank to do something else"
let progname = readLine(stdin)
if progname == "":
  quit(0)
else:
  let entry = "|" & now().format("M/dd/yyyy h:mmtt") & "|" & progname & "|\n"
  if "\n\n" in log:
    log.insert(entry, log.find("\n\n") + 1)
  else: log.add(entry)
  logfile.writeFile(log)

File: install_logger.py
import datetime

logfile = "Log4.md"

try:
    with open(logfile, "r") as file:
        log = file.read()
except FileNotFoundError:
    log = "|Timestamp|Software|\n|-|-|\n"

print("Enter program name to log it. Leave blank to do something else")
progname = input()

if progname == "":
    quit(0)
else:
    entry = "|" + datetime.datetime.now().strftime("%m/%d/%Y %H:%M%p") + "|" + progname + "|\n"
    if "\n\n" in log:
        log = log[:log.find("\n\n")+1] + entry + log[log.find("\n\n")+1:]
    else:
        log += entry

    with open(logfile, "w") as file:
        file.write(log)